src/scripts/main.js

```diff
@@ -1,3 +1,251 @@
+  return +salary.slice(1).replaceAll(',', '');
+}
+
+function printNewTable(newEmployees) {
+  tbody.innerHTML = '';
+
+  newEmployees.forEach(emp => tbody.append(emp));
+}
+
+thead.addEventListener('click', e => {
+  desc = query === e.target.innerText ? !desc : false;
+
+  query = e.target.innerText;
+
+  const newEmployees = sortEmployees(QUERY[query]);
+
+  printNewTable(newEmployees);
+});
+
+[...employees].forEach(emp => emp.addEventListener('click', () => {
+  [...employees].forEach(e => {
+    e.classList = '';
+  });
+
+  emp.classList.add('active');
+}));
+
+const form = document.createElement('form');
+
+form.classList = 'new-employee-form';
+
+form.innerHTML = `
+<label>Name: <input name="name" type="text" data-qa="name"></label>
+<label>Position: <input name="position" type="text" data-qa="position"></label>
+<label>Office:
+<select name="office" data-qa="office">
+<option></option>
+<option value="Tokyo">Tokyo</option>
+<option value="Singapore">Singapore</option>
+<option value="London">London</option>
+<option value="New York">New York</option>
+<option value="Edinburgh">Edinburgh</option>
+<option value="San Francisco">San Francisco</option>
+</select>
+</label>
+<label>Age: <input name="age" type="number" data-qa="age"></label>
+<label>Salary: <input name="salary" type="number" data-qa="salary"></label>
+<button>Save to table</button>
+`;
+
+body.append(form);
+
+form.addEventListener('click', e => {
+  const target = e.target;
+  const button = form.querySelector('button');
+
+  if (target !== button) {
+    return;
+  }
+
+  e.preventDefault();
+
+  const formName = form.children[0].children[0].value;
+  const formPosition = form.children[1].children[0].value;
+  const formOffice = form.children[2].children[0].value;
+  const formAge = form.children[3].children[0].value;
+  const formSalary = form.children[4].children[0].value;
+
+  if (!formPosition.trim() || !formOffice || !formSalary) {
+    showErrorMessage('Error',
+      'All inputs must be filled',
+      'error');
+
+    return;
+  }
+
+  if (formName.trim().length < 4) {
+    showErrorMessage('Wrong name',
+      'Name must contains at least 4 characters',
+      'error');
+
+    return;
+  }
+
+  if (+formAge < 18 || +formAge > 90) {
+    showErrorMessage('Wrong age',
+      'Age must be more than 18 and less than 90',
+      'error');
+
+    return;
+  }
+
+  showErrorMessage('Success',
+    'Employee was created',
+    'success');
+
+  tbody.insertAdjacentHTML('afterbegin', `
+  <tr>
+    <td>${formName.trim()}</td>
+    <td>${formPosition.trim()}</td>
+    <td>${formOffice}</td>
+    <td>${formAge}</td>
+    <td>$${formSalary.replace(/\B(?=(\d{3})+(?!\d))/g, ',')}</td>
+  </tr>
+  `);
+
+  form.reset();
+});
+
+function showErrorMessage(title, description, type) {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification');
+  notification.setAttribute('data-qa', 'notification');
+
+  switch (type) {
+    case 'success':
```

Comment:
Consider will be better if you create a consntants for magic words and use it everywhere